projects/Mallard/ios/Mallard/ExternalLink/ExternalLink.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  ExternalLink.swift
+//  Mallard
+//
+//  Created by James Miller on 22/09/2022.
+//  Copyright © 2022 Facebook. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@available(iOS 13.0.0, *)
+@objc(ExternalLink)
+class ExternalLink: NSObject {
+
+  @objc func canOpen() {
+    if #available(iOS 16.0, *) {
+      Task {
+        let result = try await ExternalLinkAccount.canOpen
+        return result
+      }
+    }
+   }
```

Comment:
Well, the above comments are more FYI rather than something that will actually help with your problem of interfacing React Native with Swift's Async/Await. I found this SO post which suggested using events (https://stackoverflow.com/questions/72302077/react-native-module-in-swift-with-async-function-throws-is-not-a-recognized-obje), would that be an option, do you think? Something like this:

```
  @objc
    func canOpen() {
        guard #available(iOS 16.0, *) else { return }
        Task {
            let result = await ExternalLinkAccount.canOpen
            sendEvent(withName: "OpenEvent", body: result)
        }
    }
```

I'm not sure how you'd get the React layer to await the result from the event, but maybe this is an option? It sounds like Swift `async` functions don't work in React Native, which is a shame as many new iOS APIs are Swift-only and often use async/await...